ADR/ADR-001-Git-Branching-Model.md

```diff
@@ -0,0 +1,225 @@
+
+Git Flow - The branching model by Vincent Driessen using both main/master and develop branches
+GitHub Flow - GitHub's lightweight branch-based workflow using only main/master
+
+The Git Flow model has been so far the choice for many operational software packages, providing a structured approach to managing releases and features, as well as support branches for hotfixes and long-term maintenance. It has allowed for parallel development of features while maintaining a stable main branch for production releases.
+
+Meanwhile, GitHub Flow has gained popularity for its simplicity and alignment with continuous delivery practices, especially in web applications. It allows for rapid development and deployment cycles, but requires strong discipline to ensure the main branch remains stable and deployable.
+
+We recognise that the choice of branching strategy significantly impacts development velocity, release management, code stability, and team coordination. Different repositories may have varying complexity levels and operational requirements, necessitating a flexible approach that can evolve with project needs.
+
+**Decision Drivers**
+
+* Development team size and experience levels
+* Release frequency and deployment patterns
+* Code review and quality assurance processes
+* Operational complexity and risk tolerance
+* Repository maturity and usage patterns
+* Need for supporting multiple software versions
+
+### Options Considered
+
+#### Option 1: Git Flow (Vincent Driessen's Model)
+
+Based on the branching model described at https://nvie.com/posts/a-successful-git-branching-model/
+
+**Structure**
+
+* main/master: Production-ready code, tagged releases
+* develop: Integration branch for next release containing latest delivered development changes
+* feature branches: Individual feature development, branched from develop
+* release branches: Preparation for production releases, branched from develop
+* hotfix branches: Critical production fixes, branched from main/master
+* support branches: Long-term maintenance for older releases
+
+**Advantages**
+
+* Clear separation of concerns: Production code (main) vs. development integration (develop)
+* Stable main branch: Always reflects production state
+* Structured release process: Release branches allow final testing and bug fixes
+* Parallel development: Multiple features can be developed simultaneously
+* Better for complex operations: Provides safety nets for high-risk deployments
+* Version management: Excellent for maintaining multiple software versions
+* Audit trail: Clearer history of what went into each release
+* Rollback safety: Easy to identify last known good state
+* Suited for scheduled releases: Works well with planned release cycles
+
+**Disadvantages**
+
+* Increased complexity: More branches to manage and understand
+* Merge overhead: Additional merge operations between develop and main
```

Comment:
In a workflow with specified approvers, who are not necessarily the people doing the work, this means that the final merges are invariably done using the command line tools, even if review/etc. is done using the web PR infrastructure.

The workflow doesn't feel clean.

Reply:
Why do we need to use the command line tools here? We're not talking about using the gitflow utilities here but following the git flow model, which can work using the GitHub web PR infrastructure.